File: javascript/src/controller/EnrollmentController.js
/**
 * Enrollment Controller - handles requests and calls service layer
 * This is simplified - in real apps, this would handle HTTP requests
 */
class EnrollmentController {
    constructor(enrollmentService) {
        this.enrollmentService = enrollmentService;
    }

    /**
     * Enroll a student in a course
     */
    enrollStudent(studentId, courseId) {
        try {
            const enrollment = this.enrollmentService.enroll(studentId, courseId);
            return `SUCCESS: Student ${studentId} enrolled in course ${courseId}`;
        } catch (error) {
            return `ERROR: ${error.message}`;
        }
    }

    /**
     * Get student GPA
     */
    getStudentGPA(studentId) {
        try {
            const gpa = this.enrollmentService.calculateGPA(studentId);
            return `GPA for student ${studentId}: ${gpa.toFixed(2)}`;
        } catch (error) {
            return `ERROR: ${error.message}`;
        }
    }
}

module.exports = EnrollmentController;

File: python/src/controller/enrollment_controller.py
"""
Enrollment Controller - handles requests and calls service layer
This is simplified - in real apps, this would handle HTTP requests
"""
from ..service.enrollment_service import EnrollmentService


class EnrollmentController:
    def __init__(self, enrollment_service: EnrollmentService):
        self.enrollment_service = enrollment_service
    
    def enroll_student(self, student_id: str, course_id: str) -> str:
        """Enroll a student in a course"""
        try:
            enrollment = self.enrollment_service.enroll(student_id, course_id)
            return f"SUCCESS: Student {student_id} enrolled in course {course_id}"
        except Exception as e:
            return f"ERROR: {str(e)}"
    
    def get_student_gpa(self, student_id: str) -> str:
        """Get student GPA"""
        try:
            gpa = self.enrollment_service.calculate_gpa(student_id)
            return f"GPA for student {student_id}: {gpa:.2f}"
        except Exception as e:
            return f"ERROR: {str(e)}"
